NeosModLoader/Utility/LaunchArguments.cs

```diff
@@ -0,0 +1,354 @@
+using FrooxEngine;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static FrooxEngine.FinalIK.IKSolverVR;
+
+namespace NeosModLoader.Utility
+{
+	/// <summary>
+	/// Contains methods to access the command line arguments Neos was launched with.
+	/// </summary>
+	/// Use analyze on Environment.GetCommandLineArgs() to find all possible arguments.
+	public static class LaunchArguments
+	{
+		/// <summary>
+		/// Prefix symbol that indicates an argument name: <c>-</c>
+		/// </summary>
+		public const char ArgumentIndicator = '-';
+
+		/// <summary>
+		/// Prefix string after the argument indicator, that indicated an argument targeted at NML itself or mods loaded by it: <c>NML.</c><br/>
+		/// This is removed from argument names to get their proper name.
+		/// </summary>
+		public const string NMLArgumentPrefix = "NML.";
+
+		private static readonly Dictionary<string, Argument> arguments = new(StringComparer.InvariantCultureIgnoreCase);
+
+		private static readonly string[] possibleNeosExactFlagArguments =
+		{
+			"ctaa", "ctaatemporaledgepower", "ctaasharpnessenabled", "ctaaadaptivesharpness", "ForceSRAnipal",
+			"StereoDisplay", "MixedReality", "DirectComposition", "ExternalComposition", "create_mrc_config",
+			"load_mrc_config"
+		};
+
+		private static readonly string[] possibleNeosExactParameterArguments =
+		{
+			"TextureSizeRatio", "DataPath", "CachePath"
+		};
+
+		private static readonly string[] possibleNeosInfixFlagArguments =
+		{
+			"CameraBiggestGroup", "CameraTimelapse", "CameraStayBehind", "CameraStayInFront", "AnnounceHomeOnLAN"
+		};
+
+		private static readonly string[] possibleNeosSuffixFlagArguments =
+		{
+			"GeneratePrecache", "Verbose", "pro", "UseLocalCloud", "UseStagingCloud", "ForceRelay", "Invisible",
+			"ForceReticleAboveHorizon", "ForceNoVoice", "ResetDash", "Kiosk", "DontAutoOpenCloudHome", "ResetUserspace",
+			"ForceLANOnly", "DeleteUnsyncedCloudRecords", "ForceSyncConflictingCloudRecords", "ForceIntroTutorial",
+			"SkipIntroTutorial", "RepairDatabase", "UseNeosCamera", "LegacySteamVRInput", "Etee", "HideScreenReticle",
+			"Viveport", "DisableNativeTextureUpload"
+		};
+
+		private static readonly string[] possibleNeosSuffixParameterArguments =
+		{
+			"Config", "LoadAssembly", "BackgroundWorkers", "PriorityWorkers", "Bench", "Watchdog", "Bootstrap",
+			"OnlyHost", "Join", "Open", "OpenUnsafe", "EnableOWO"
+		};
```

Comment:
I don't think it's a good idea to rely on a list of Neos arguments. They can and will be changed if Neos ever gets updated again. NML should not be concerned with these, and if a mod is concerned with one, they could easily implement their own check for it IMO.

Reply:
I wanted to do only `-NML.` ones first, but then I was like... it'd be kinda nice to know if an upcoming one is part of the Neos ones - and then I might as well just expose them. They're also important to know for mod creators wanting to add their own, as there's a lot of them that block any other argument ending with them or even containing them